Compiler/src/implementation/Scanner.cpp

```diff
@@ -196,6 +200,72 @@ Token Scanner::consumeZeroToken()
 	return Token(INTEGER, "0");
 }
 
+Token Scanner::consumeGenericWordToken()
+{
+	string Accumulator = string(1, FileContents[FilePosition]);
+
+	//Set initial endState To False
+	bool ValidEndState = false;
+	while (!ValidEndState && FilePosition < FileContents.size())
+	{
+		if (FilePosition + 1 == FileContents.size())
+		{
+			//Last character in file. Update accumulator, and set End State To Valid.
+			FilePosition++;
+			ValidEndState = true;
+			continue;
+		}
+
+		char NextChar = FileContents[FilePosition + 1];
+
+		//If enot the last character in file, check the next char.
+		//Also keep making sure that Accumulator doesn't exceed more than 256 characters.
+		if (isalpha(NextChar) && Accumulator.length() < 257) {
```

Comment:
So right now if the accum is > 257 we throw the same error as if they had an unexpected character. Should we move this completely towards the end. So like, we get to a valid stop state, then we check the length. If invalid throw custom error : something like "This identifier is too long!"? That way we have different identifiers. Same concept for integers, do we want a custom error message for those cases? I feel like it's valuable enough to do.

Reply:
Also, I'll tackle these merge conflicts tonight when I'm done with class. Should be straightforward enough.